src/app/core/authentication/authentication.service.ts

```
@@ -73,25 +74,74 @@ export class AuthenticationService {
+    return Math.max(0, tokenExpiry - now);
+  }
+
+  private setupTokenRefresh(timeUntilExpiry: number): void {
+    if (timeUntilExpiry > 60000) { // If more than 1 minute left
+      // Token is still valid, set up refresh
+      this.refreshTokenOnExpiry(Math.floor(timeUntilExpiry / 1000));
+      this.refreshAccessToken = false; // Token is still valid
     }
   }
 
   hasAccess(permission: String): Boolean {
```

Biome (1.9.4)

[error] Don't use 'String' as a type.

Use lowercase primitives for consistency.
Safe fix: Use 'string' instead


(lint/complexity/noBannedTypes)